src/plugins/network.ts

```diff
@@ -107,6 +111,23 @@ export class NetworkManager {
     });
   }
 
+  _onRequestWillBeSentExtraInfo(
+    event: Protocol.Network.requestWillBeSentExtraInfoPayload
+  ) {
+    const { requestId, headers } = event;
+    const record = this.waterfallMap.get(requestId);
+    if (!record) {
+      return;
+    }
+    /**
+     * Enhance request headers with additional information
+     */
+    record.request.headers = {
```

Comment:
this is done here instead of on response as `response` event can sometimes be not fired for dropped requests, so moved it before. Also this lets us do our next iteration in our waterfall which is `Add flyout in waterfall`